src/dftbp/timedep/linrespcommon.F90

```diff
@@ -411,469 +406,464 @@ subroutine getSqrOcc(occNr, win, nmatup, nmat, getIA, tSpin, n_ij)
+            end if
+            getABasym(abs, 1) = aa
+            getABasym(abs, 2) = bb
+            getABasym(abs, 3) = ss
+          end do
+        end do
+      end do
+
+      call distributeRangeInChunks(env, 1, nxvv_a, iGlobalAB, fGlobalAB)
+      nLocAB = fGlobalAB - iGlobalAB + 1
 
-    if (tHybridXc) then
-      ! Number of vir-vir transitions a->b _and_ b->a, summed over spin channels
-      nxvv_a = sum(nvir_ud**2)
-      allocate(qv(natom, max(sum(nxov_ud), nxvv_a)))
+      allocate(qv(lr%nAtom, max(nLoc, nLocAB)))
+      allocate(vGlb(rpa%nxov_rd))
+      allocate(vGlb2(rpa%nxov_rd))
+
+      !> TD-LC-DFTB seems to require two additional global arrays of dim nOcc*nVir, qv is local 
       qv(:,:) = 0.0_dp
-      do jbs = 1, nmat
-        jj = getIA(win(jbs), 1)
-        bb = getIA(win(jbs), 2)
-        ss = getIA(win(jbs), 3)
-        do aa = nocc_ud(ss) + 1, norb
-          abs = iaTrans(aa, bb, ss)
-          qij(:) = transChrg%qTransAB(abs, env, denseDesc, ovrXev, grndEigVecs, getAB)
-          jas = iaTrans(jj, aa, ss)
-          qv(:,jas) = qv(:,jas) + qij * vTmp(jbs) * sqrOccIA(jbs)
+      vGlb(:) = 0.0_dp
+      vGlb2(:) = 0.0_dp
+
+      !> Gather local arrays in corresponding global array
+      call gatherChunks(env, 1, rpa%nxov_rd, vLoc, vGlb)
+
+      !> Compute w_ia = q^ij_A GLR_AB q^ab v_jb
+      do jas = iGlobal, fGlobal
+        myja = jas - iGlobal + 1
+        jj = rpa%getIA(rpa%win(jas), 1)
+        aa = rpa%getIA(rpa%win(jas), 2)
+        ss = rpa%getIA(rpa%win(jas), 3)
+        do bb = rpa%nocc_ud(ss) + 1, nOrb
+          abs = rpa%iaTrans(aa, bb, ss)
+          qij(:) = transChrg%qTransAB(abs, env, denseDesc, ovrXev, grndEigVecs, rpa%getAB)
+          jbs = rpa%iaTrans(jj, bb, ss)
+          qv(:,myja) = qv(:,myja) + qij(:) * vGlb(jbs) * rpa%sqrOccIA(jbs)
         end do
-      end do
-      ! Attention: Necessary to diff. between nmat/nxov_rd and all possible exc
-      do jas = 1, sum(nxov_ud)
-        otmp(:) = qv(:,jas)
-        call hemv(qv(:,jas), lrGamma, otmp)
-      end do
-      do ias = 1, nmat
-        ii = getIA(win(ias), 1)
-        aa = getIA(win(ias), 2)
-        ss = getIA(win(ias), 3)
-        do jj = 1, nocc_ud(ss)
-          ijs = iaTrans(ii, jj, ss)
-          jas = iaTrans(jj, aa, ss)
-          qij(:) = transChrg%qTransIJ(ijs, env, denseDesc, ovrXev, grndEigVecs, getIJ)
-          vout(ias) = vout(ias) - cExchange * sqrOccIA(ias) * dot_product(qij, qv(:, jas))
+        
+        otmp(:) = qv(:,myja)
+        call hemv(qv(:,myja), lrGamma, otmp, uplo='U')
+        
+        do ii = 1, rpa%nocc_ud(ss)
+          ijs = rpa%iaTrans(ii, jj, ss)
+          qij(:) = transChrg%qTransIJ(ijs, env, denseDesc, ovrXev, grndEigVecs, rpa%getIJ)
+          ias = rpa%iaTrans(ii, aa, ss)
+          vGlb2(ias) = vGlb2(ias) - cExchange * rpa%sqrOccIA(ias) * dot_product(qij, qv(:,myja))
         end do
       end do
 
+      !> Compute w_ia = q^ib_A GLR_AB q^ja v_jb 
```

Comment:
```suggestion
      ! Compute w_ia = q^ib_A GLR_AB q^ja v_jb 
```
Note, `!>` is only used to indicate documentation comments of interfaces (procedures, arguments). For commenting code we use the simple `!`.